src/modules/swaps/swaps.repository.ts

```diff
@@ -193,4 +202,29 @@ export class SwapRepository {
       },
     };
   }
+
+  private async getSwapStats(
+    queryBuilder: SelectQueryBuilder<Swap>,
+    assetId: string,
+  ): Promise<SwapsStatsDto> {
+    const [data] = await queryBuilder.getManyAndCount();
+
```

Comment:
Pass `data ` as a parameter to function instead of `queryBuilder`.

```suggestion
const length = data.length;
const buys = data.filter(swap => swap.outputAssetId === assetId).length;
const sells = length - buys;

return {
      buys,
      sells,
    };
```

Reply:
@DanijelCBS I believe that what @sdjuka needs is amount and not count of buys/sells.